### ISSUE TITLE
Bundler's inline `install` option misdocumented

### ISSUE THREAD

**Issue by ngan** [rubygems/rubygems#5364]
According to the [source documentation](https://github.com/rubygems/rubygems/blob/master/bundler/lib/bundler/inline.rb#L11-L13), the `install` argument specifies whether or not installation of missing gems should happen or not. However, looking at the code, it looks like installation will [_always_ happen for missing gems](https://github.com/rubygems/rubygems/blob/master/bundler/lib/bundler/inline.rb#L61). The only thing that the flag does is determines whether to [quiet the installation message or not](https://github.com/rubygems/rubygems/blob/master/bundler/lib/bundler/inline.rb#L60).

Considering that it already [recognizes a `quiet` option](https://github.com/rubygems/rubygems/blob/master/bundler/lib/bundler/inline.rb#L37), perhaps we should just fix the logic to:

```ruby
if install && definition.missing_specs?
```

**Comment by ngan** [comment]
@deivid-rodriguez does this make sense? Would be happy to open PR.

**Comment by deivid-rodriguez** [comment]
@ngan Sorry I never left feedback here.

Since `gemfile(&gemfile)` already behaves essentially the same as `gemfile(true, &gemfile)`, wouldn't it make more sense to deprecate the `install` parameter, since `bundler/inline` installs what's necessary regardless?

The patch you suggest changes (misdocumented, that's true) behavior because people used to `gemfile(&gemfile)` auto-installing as necessary will no longer get that behavior.

**Comment by tonyaraujop** [comment]
Hello there @ngan have you started to work on this? If not, I would like to help.

**Comment by ngan** [comment]
Hey @tonyaraujop I started this but didn't have time to finish. It's fairly straightforward. I think where I got stuck was how to how to show the deprecation message. Summer distract me 😓 

Feel free to have a go at it!

**Comment by tonyaraujop** [comment]
Thanks.

**Comment by tonyaraujop** [comment]
Hey @ngan I looked in files and found `deprecate.rb` under `lib/rubygems` that has a method `rubygems_deprecate` which `Indicate something will be removed in the next major RubyGems version, and (optionally) a replacement for it.` So I just added this line `rubygems_deprecate(install)` just under the line `definition.validate_runtime!`. Do you think this is enough? I tried to test in `inline_spec.rb` and when I use `rake test`, all tests passes. I feel that something is wrong but I can tell where.

**Comment by jeromedalbert** [comment]
I can confirm that this is still an issue: there is a discrepancy in documentation and actual behavior, as explained by ngan.

**Comment by deivid-rodriguez** [comment]
After looking into the implementation today, I understood what `install` really controls. 

* `install = false` behaves like `bundle install --prefer-local`, i.e., uses locally available gems if they satisfy inline gemfile requirements, and otherwise resolves remotely to the latest resolvable versions, and installs what's necessary.
* `install = true` always resolves remotely regardless of what's present locally, so that if new versions of inline dependencies have been released, they will always get installe

So basically, even if both install gems if necessary, `install = false` is more conservative while `install = true` always prefers to install the latest version of dependencies even if the gemfile is satisfiable locally.

I think this parameter makes sense after all, so I would suggest to rename it and reword documentation to be more clear.
